src/main/kotlin/no/digdir/fdk/rdf/parse/eventpublisher/kafka/KafkaReasonedEventConsumer.kt

```diff
@@ -46,35 +49,54 @@ class KafkaReasonedEventConsumer(
     fun listen(record: ConsumerRecord<String, SpecificRecord>, ack: Acknowledgment) {
         LOGGER.debug("Received message - offset: " + record.offset())
         val event = record.value()
+
+        val resourceType = when (event) {
+            is DatasetEvent -> RdfParseResourceType.DATASET
+            is DataServiceEvent -> RdfParseResourceType.DATA_SERVICE
+            is ConceptEvent -> RdfParseResourceType.CONCEPT
+            is InformationModelEvent -> RdfParseResourceType.INFORMATION_MODEL
+            is ServiceEvent -> RdfParseResourceType.SERVICE
+            is EventEvent -> RdfParseResourceType.EVENT
+            else -> null
+        }
+
         try {
             event.let {
                 if (it is DatasetEvent && it.type == DatasetEventType.DATASET_REASONED) {
-                    parseAndProduce(it.fdkId.toString(), it.graph.toString(), it.timestamp, RdfParseResourceType.DATASET)
+                    parseAndProduce(it.fdkId.toString(), it.graph.toString(), it.timestamp, resourceType!!)
```

Comment:
Kan `resourceType` gjøres non-nullable i stedet for? Og hvis den må være nullable, er det ikke bedre å sjekke og håndtere hvis den er null?